### PR TITLE
Navigation Bar: Drops custom metrics

### LINKED ISSUE #852: Editor: RTL is not respected by menu buttons and navigation
When using Simplenote in a right-to-left language such as Arabic, menu items in the note editor are still positioned on the right and the back button is still positioned on the left, with the indicator now facing the wrong direction.

The app respects RTL on the top-level note list view and correctly handles the menu too. It only seems to affect the note editor navigation bar

### Expected
Menu icons should be in reverse order on the left hand side of the navigation bar. The back button should be on the right with the back arrow pointing to the right

### Observed
Menu icons are still on the right hand side in LTR order and the back button is still on the left, pointing to the left.

### Reproduced
1. Switch iPad/iPhone to Arabic language and region to Saudi Arabia
2. Open a note in Simplenote
3. Observe that menu icons are on the top right

Simulator Screen Shot - iPad Pro (12 9-inch) (3rd generation) - 2020-08-20 at 17 24 45

Make|Model|iOS Version|App Version
-|-|-|-
Apple|iPad Pro 12.9 Inch|iPadOS 13.5|4.20.1

### LINKED ISSUE #918: Navigation Bar Adjustments
### Details:
- [x] Drop custom padding / navbar shrinking
- [x] Drop custom Navigation Bar Views
- [x] Adjust the back text to say All Notes (or the current tag selection)

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
In this PR we're dropping all of the customized Navigation Bar metrics. All of them!

@eshurakov Sir!! May I bug you with this one?  (Thaaank youuu!)

@SylvesterWilmott Having all of the default OS animations within the Navigation Title feels **so perfect!!**

Closes #918
Closes #852

### Test: List
1. Access the Notes List

- [x] Verify on the top left corner there's a navigation button that allows you to access the Sidebar
- [x] Verify on the top right corner there's a navigation button that allows you to add a new Note

### Test: Editor
1. Access the Notes List
2. Press over any note

- [x] Verify the top left navigation button allows you to go back to the Notes List
- [x] Verify that the `( i )` navigation button presents the Note Options
- [x] Verify that the `+` navigation button adds a new Note
- [x] Verify that pressing anywhere causes the top right Bar Buttons to refresh: Add Checklist and Dismiss Keyboard should show up

### Test: Markdown
1. Access the Notes List
2. Press over any note
3. Enable Markdown Preview
4. Slide from the right to push the Markdown Preview UI

- [x] Verify the top left navigation button allows you to go back

### Test: Blank Notes
1. Access the Notes List
2. Press over the top right `+` icon
3. Don't type anything!
4. Press back

- [x] Verify the newly created note isn't anywhere to be found in the Notes List!

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/Classes/SPNoteEditorViewController.m [hunk 1]
```diff
@@ -567,30 +392,19 @@ - (void)prepareToPopView
                                             atScrollPosition:UITableViewScrollPositionTop
                                                     animated:NO];
     }
-    
 }
 
-
 - (void)backButtonAction:(id)sender
```
Comment (eshurakov) [inline Simplenote/Classes/SPNoteEditorViewController.m]: `backButtonAction` is not triggered by back button. I see that it is used when deleting note from options or when note is deleted on another device. Also from the "pop" side gesture.
So maybe it makes sense to rename it to `dismiss` or something along those lines.

**Review by eshurakov — COMMENTED** [review]
Looks good! 👍

**Comment by eshurakov** [comment]
Found a small issue (in the current live app it works):
- Create a new note
- Quickly type something
- Tap back (before the timer to save the note is executed)
- Note is lost

**Comment by jleandroperez** [comment]
@eshurakov Nice catch, thank you!!

Ready for another look, all comments addressed 🔥

**Review by eshurakov — APPROVED** [review]
🚢  :shipit:

**Comment by jleandroperez** [comment]
Thank you sir!!

**Pull request merged**